### PR TITLE
Enable aten.upsample_nearest2d.vec

### PR DESCRIPTION
**Pull request by cccclai**
Summary:
As title, fix some edge cases and add two unit tests
```
buck test 'fbcode//mode/dev-nosan' fbcode//executorch/backends/qualcomm/tests/fb:test_qnn_delegate_simulator -- --exact 'executorch/backends/qualcomm/tests/fb:test_qnn_delegate_simulator - test_qnn_backend_up_sampling_nearest_2d_with_size (executorch.backends.qualcomm.tests.fb.test_qnn_delegate_simulator.TestQNNQuantizedOperatorSimulator)'
buck test 'fbcode//mode/dev-nosan' fbcode//executorch/backends/qualcomm/tests/fb:test_qnn_delegate_simulator -- --exact 'executorch/backends/qualcomm/tests/fb:test_qnn_delegate_simulator - test_qnn_backend_up_sampling_nearest_2d_with_size (executorch.backends.qualcomm.tests.fb.test_qnn_delegate_simulator.TestQNNFloatingPointOperatorSimulator)'
```

Differential Revision: D70796006

### CODE REVIEW

File: backends/qualcomm/tests/test_qnn_delegate.py [hunk 1]
```diff
@@ -1430,6 +1440,16 @@ def test_qnn_backend_interpolate_nearest_2d(self):
         module = self.get_qdq_module(module, sample_input)
         self.lower_module_and_test_output(module, sample_input)
 
+    def test_qnn_backend_up_sampling_nearest_2d_with_scale_factor(self):
+        module = UpsampleNearest2D(scale_factor=2)
+        sample_input = (torch.randn(1, 16, 72, 104), )
+        self.lower_module_and_test_output(module, sample_input)
```
Comment (haowhsu-quic) [inline backends/qualcomm/tests/test_qnn_delegate.py:1446]: I guess you might need this for quantized operator:
```
module = self.get_qdq_module(module, sample_input)
```

Comment (cccclai) [reply backends/qualcomm/tests/test_qnn_delegate.py:1447]: thanks - updated

**Review by haowhsu-quic — APPROVED** [review]

**Review by kirklandsign — APPROVED** [review]
